### PR TITLE
Fixture env

### PR DESCRIPTION
**Pull request by stingion**

### CODE REVIEW

File: scalecube-configuration/src/test/java/io/scalecube/configuration/scenario/BaseScenario.java [hunk 1]
```diff
@@ -12,34 +16,50 @@
...
+  protected static final Duration TIMEOUT = Duration.ofSeconds(1);
   static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
-  static final Token AUTH0_TOKEN = new Token("auth0_token");
-  static final String ORGANIZATION_1 = "Test Organization 1";
-  static final String ORGANIZATION_2 = "Test Organization 2";
+
+  protected static final Token AUTH0_TOKEN = new Token(
+      ((Supplier<String>) () -> {
+        AuthAPI authAPI = new AuthAPI("itestsoauth.eu.auth0.com",
+            "FbUh6PxUVaH1ZSy7Of03gSu7323NpiZF",
```
Comment (aharonha) [inline scalecube-configuration/src/test/java/io/scalecube/configuration/scenario/BaseScenario.java]: is (or _was_) it a secret?

File: scalecube-configuration/src/test/java/io/scalecube/configuration/scenario/BaseScenario.java [hunk 2]
```diff
@@ -63,9 +84,9 @@ private ApiKey getApiKey(
             .block(TIMEOUT);
 
     return Stream.of(
-            Optional.ofNullable(organization)
-                .orElseThrow(() -> new IllegalStateException("Organization is null"))
-                .apiKeys())
+        Optional.ofNullable(organization)
+            .orElseThrow(() -> new IllegalStateException("Organization is null"))
```
Comment (aharonha) [inline scalecube-configuration/src/test/java/io/scalecube/configuration/scenario/BaseScenario.java:101]: Is it possible to get to this point that you throw this exception?

File: scalecube-configuration/src/test/java/io/scalecube/configuration/fixtures/ContainersConfigurationServiceFixture.java [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+                      PortBinding.parse(RS_GATEWAY_PORT + ":" + RS_GATEWAY_PORT)
+                  );
+                })
+            .waitingFor(new HostPortWaitStrategy());
+    gatewayContainer.start();
+  }
+
+  private void setUpOrganizationService(Map<String, String> env) {
+    env.put("JAVA_OPTS", "-Dio.scalecube.organization.seeds=" + GATEWAY_NETWORK_ALIAS
+        + ":4801 -DmockTokenVerifier=true");
```
Comment (dmytro-lazebnyi) [inline scalecube-configuration/src/test/java/io/scalecube/configuration/fixtures/ContainersConfigurationServiceFixture.java]: Let's avoid using such approach: adding some mocks to repository of organization service (moreover, adding mocks to production code, https://github.com/scalecube/scalecube-organization-service/pull/135) just to use it for tests in another project (configuration service).

Set up organization service without Docker container and mock anything you need.

**Review by dmytro-lazebnyi — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 4]
```diff
@@ -86,6 +110,11 @@
         <artifactId>scalecube-services-api</artifactId>
         <version>${scalecube-services.version}</version>
       </dependency>
+      <dependency>
+        <groupId>io.scalecube</groupId>
+        <artifactId>scalecube-services-gateway-clientsdk</artifactId>
+        <version>${scalecube-services.version}</version>
```
Comment (dmytro-lazebnyi) [inline pom.xml:116]: Shouldn't it be in `test` scope?

File: scalecube-configuration/pom.xml [hunk 5]
```diff
@@ -129,6 +147,23 @@
       <version>${commons-lang3.version}</version>
       <scope>test</scope>
     </dependency>
+
+    <dependency>
+      <groupId>org.testcontainers</groupId>
+      <artifactId>couchbase</artifactId>
+      <version>${testcontainers.version}</version>
```
Comment (dmytro-lazebnyi) [inline scalecube-configuration/pom.xml:155]: `test` scope?

File: scalecube-configuration/pom.xml [hunk 6]
```diff
@@ -129,6 +147,23 @@
...
+    </dependency>
+    <dependency>
+      <groupId>org.testcontainers</groupId>
+      <artifactId>vault</artifactId>
+      <version>${testcontainers.version}</version>
```
Comment (dmytro-lazebnyi) [inline scalecube-configuration/pom.xml:161]: `test` scope?

File: scalecube-configuration/pom.xml [hunk 7]
```diff
@@ -129,6 +147,23 @@
...
+    </dependency>
+    <dependency>
+      <groupId>com.auth0</groupId>
+      <artifactId>auth0</artifactId>
```
Comment (dmytro-lazebnyi) [inline scalecube-configuration/pom.xml]: Is this dependency somewhere used?

File: scalecube-configuration/src/test/java/io/scalecube/configuration/CreateRepositoryIT.java [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+@WithFixture(value = ContainersConfigurationServiceFixture.class, lifecycle = Lifecycle.PER_METHOD)
+public class CreateRepositoryIT extends CreateRepositoryScenario {
+
+  public CreateRepositoryIT() {
+    super(new ITInitBase());
+  }
+
+  @Disabled
+  @Override
+  protected void createRepositoryUsingExpiredToken(
```
Comment (dmytro-lazebnyi) [inline scalecube-configuration/src/test/java/io/scalecube/configuration/CreateRepositoryIT.java:20]: Please, do not disable tests.

File: scalecube-configuration/src/test/java/io/scalecube/configuration/DeleteEntryIT.java [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+
+  public DeleteEntryIT() {
+    super(new ITInitBase());
+  }
+
+  @Disabled
+  @Override
+  protected void deleteEntryUsingExpiredToken(
+      ConfigurationService configurationService, OrganizationService organizationService) {
+  }
```
Comment (dmytro-lazebnyi) [inline scalecube-configuration/src/test/java/io/scalecube/configuration/DeleteEntryIT.java:22]: Please, do not disable tests.

File: scalecube-configuration/src/test/java/io/scalecube/configuration/FetchEntriesIT.java [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+
+  public FetchEntriesIT() {
+    super(new ITInitBase());
+  }
+
+  @Disabled
+  @Override
+  protected void fetchEntriesUsingExpiredToken(
+      ConfigurationService configurationService, OrganizationService organizationService) {
+  }
```
Comment (dmytro-lazebnyi) [inline scalecube-configuration/src/test/java/io/scalecube/configuration/FetchEntriesIT.java:22]: Please, do not disable tests.

File: scalecube-configuration/src/test/java/io/scalecube/configuration/FetchEntryIT.java [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+
+  public FetchEntryIT() {
+    super(new ITInitBase());
+  }
+
+  @Disabled
+  @Override
+  protected void fetchEntryUsingExpiredToken(
+      ConfigurationService configurationService, OrganizationService organizationService) {
+  }
```
Comment (dmytro-lazebnyi) [inline scalecube-configuration/src/test/java/io/scalecube/configuration/FetchEntryIT.java:22]: Please, do not disable tests.

File: scalecube-configuration/src/test/java/io/scalecube/configuration/SaveEntryIT.java [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+
+  public SaveEntryIT() {
+    super(new ITInitBase());
+  }
+
+  @Disabled
+  @Override
+  protected void saveEntryUsingExpiredToken(
+      ConfigurationService configurationService, OrganizationService organizationService) {
+  }
```
Comment (dmytro-lazebnyi) [inline scalecube-configuration/src/test/java/io/scalecube/configuration/SaveEntryIT.java:22]: Please, do not disable tests.

File: scalecube-configuration/src/test/java/io/scalecube/configuration/fixtures/InMemoryConfigurationServiceFixture.java [hunk 13]
```diff
@@ -103,7 +103,7 @@ private OrganizationService mockOrganization() {
     Builder organization1InfoBuilder =
         OrganizationInfo.builder()
             .id(organizationId1)
-            .name("Test Organization 1")
+            .name("Test_Organization_Name_1")
```
Comment (dmytro-lazebnyi) [inline scalecube-configuration/src/test/java/io/scalecube/configuration/fixtures/InMemoryConfigurationServiceFixture.java:106]: What is the purpose of this change?

File: scalecube-configuration/src/test/java/io/scalecube/configuration/scenario/BaseScenario.java (hunk 1 above)
Comment (stingion) [reply scalecube-configuration/src/test/java/io/scalecube/configuration/scenario/BaseScenario.java]: Old version (removed)

File: scalecube-configuration/src/test/java/io/scalecube/configuration/scenario/BaseScenario.java (hunk 2 above)
Comment (stingion) [reply scalecube-configuration/src/test/java/io/scalecube/configuration/scenario/BaseScenario.java:101]: It's not necessary to catch it, all places it's treated as RuntimeException

File: scalecube-configuration/src/test/java/io/scalecube/configuration/fixtures/InMemoryConfigurationServiceFixture.java (hunk 13 above)
Comment (stingion) [reply scalecube-configuration/src/test/java/io/scalecube/configuration/fixtures/InMemoryConfigurationServiceFixture.java:106]: It's just for fast intuitive visibility of relation to name.

File: scalecube-configuration/src/test/java/io/scalecube/configuration/SaveEntryIT.java (hunk 12 above)
Comment (stingion) [reply scalecube-configuration/src/test/java/io/scalecube/configuration/SaveEntryIT.java:22]: Implementation of this integration tests took to much time for me, and can take more, so I disabled it temporary. So I decided to not include it in my integration tests.

File: scalecube-configuration/src/test/java/io/scalecube/configuration/FetchEntryIT.java (hunk 11 above)
Comment (stingion) [reply scalecube-configuration/src/test/java/io/scalecube/configuration/FetchEntryIT.java:22]: As below

File: scalecube-configuration/src/test/java/io/scalecube/configuration/FetchEntriesIT.java (hunk 10 above)
Comment (stingion) [reply scalecube-configuration/src/test/java/io/scalecube/configuration/FetchEntriesIT.java:22]: As below

File: scalecube-configuration/src/test/java/io/scalecube/configuration/DeleteEntryIT.java (hunk 9 above)
Comment (stingion) [reply scalecube-configuration/src/test/java/io/scalecube/configuration/DeleteEntryIT.java:22]: As below

File: scalecube-configuration/src/test/java/io/scalecube/configuration/CreateRepositoryIT.java (hunk 8 above)
Comment (stingion) [reply scalecube-configuration/src/test/java/io/scalecube/configuration/CreateRepositoryIT.java:20]: As below

File: scalecube-configuration/src/test/java/io/scalecube/configuration/fixtures/InMemoryConfigurationServiceFixture.java (hunk 13 above)
Comment (dmytro-lazebnyi) [reply scalecube-configuration/src/test/java/io/scalecube/configuration/fixtures/InMemoryConfigurationServiceFixture.java:106]: Not sure that understood it. Could you please explain it in details?

**Comment by dmytro-lazebnyi** [comment]
Suggest to close it and continue working on #133 
The only thing left to do in #133 is to run `testcontainers` inside Travis.
